logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/ephemeral/SelfUserSenderSelfDeletionUseCase.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.message.ephemeral
+
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.MessageRepository
+
+/**
+ * When the self user is the sender of the self deletion message, we only mark it as deleted because we are relaying on the receiver, telling us
```

Comment:
```suggestion
 * When the self user is the sender of the self deletion message, we only mark it as deleted because we are relying on the receiver, telling us
```